### PR TITLE
feat: templ templates

### PR DESCRIPTION
**Pull request by joerdav**

### CODE REVIEW

**Comment by acaloiaro** [comment]
Very cool. Thanks for taking the time to do this. 

If you don't mind my contributing to your PR, there are a couple things I'd like to do.

1. Since Hugo users seem to like simplicity over all, I'd like to retain the raw HTML and `html/template` endpoints and give `templ` its own, so there are clear and distinct examples of both.
2. Document in `README` the `templ` generation steps required to make changes to the `templ` templates. 

In any case, this is very cool and I plan to incorporate this templ approach into a current project as well! I'll wait to hear back from you before making any changes.

**Comment by joerdav** [comment]
Happy for you to make those changes yep.

File: server.go [hunk 1]
```diff
@@ -20,9 +21,25 @@ func main() {
 	// Serve all hugo content (the 'public' directory) at the root url
 	mux.Handle("/", http.FileServer(http.FS(serverRoot)))
 
+	cors := func(h http.Handler) http.HandlerFunc {
```
Comment (acaloiaro) [inline server.go:26]: Nice change!

File: server.go [hunk 2]
```diff
@@ -88,23 +78,10 @@ func helloWorldForm(w http.ResponseWriter, r *http.Request) {
 	}
 
 	name = r.FormValue("name")
-	// we're dealing with a really simple template; let's use an inline string instead of a whole separate file for this
-	// one
-	tmpl, err := template.New("form_response").Parse("<h3>Greeting: Hello, {{ .Name }}!</h3>")
-	if err != nil {
-		ise(err, w)
+	if err := partials.HelloWorldGreeting(name).Render(r.Context(), w); err != nil {
```
Comment (acaloiaro) [inline server.go:92]: :+1:

**Review by acaloiaro — APPROVED** [review]

**Pull request merged**